### PR TITLE
Fix CheckboxList UI not updating when values are set programmatically

### PR DESCRIPTION
**Pull request by readingdancer**
Fixes: https://github.com/umbraco/Umbraco-CMS/issues/19437

**Prerequisites**

- [x] I have added steps to test this contribution in the description below

This fixes the CheckboxList UI update issue where programmatically set values don't reflect in the UI until page refresh.

**Description**
What did I do?
Fixed a UI synchronization bug in the CheckboxList property editor where setting values programmatically (via JavaScript) would not update the visual state of the checkboxes until the page was refreshed.

**Why did I do it?**
The UmbPropertyEditorUICheckboxListElement had a disconnect between its internal value storage (#selection) and the UI rendering state (_list items). When the value setter was called programmatically, it updated the selection array but failed to sync the checked property of the list items that control the visual checkbox states.
This caused confusion for developers building custom property editors or integrations, as the UI would appear broken despite the underlying data being correct.

**How can we test the changes?**

1. Create test setup:

- Create a document type with a CheckboxList property
- Configure it with options: "Red", "Green", "Blue"
- Create a document using this document type

2. Test programmatic value setting:
```JAVASCRIPT
   // In browser console or custom property editor
   const checkboxElement = document.querySelector('umb-property-editor-ui-checkbox-list');
   checkboxElement.value = ["Red", "Blue"];
```

3. Verify immediate UI update:

- ✅ Expected: "Red" and "Blue" checkboxes should immediately appear checked
- ❌ Before fix: No checkboxes would appear checked until page refresh
- ✅ After fix: Checkboxes update immediately

4. Test edge cases:
- Setting empty array: element.value = [] (should uncheck all)
- Setting single value: element.value = ["Green"] (should check only Green)
- Setting invalid values: element.value = ["Invalid"] (should handle gracefully)

**Technical Implementation:**

- Added #updateCheckedState() method that:
- Maps through _list items and updates their checked property based on current #selection
- Triggers requestUpdate() to ensure Lit element re-renders
- Called from the value setter to maintain UI synchronization

**Code Changes:**

- Modified property-editor-ui-checkbox-list.element.ts
- Added 6 lines of code (method + call)
- Zero breaking changes - purely additive fix
- Maintains all existing functionality

This fix ensures the CheckboxList property editor behaves consistently whether values are set through user interaction or programmatically, improving the developer experience for custom integrations and property editors.

### CODE REVIEW

**Comment by readingdancer** [comment]
> Copilot encountered an error and was unable to review this pull request. You can try again by re-requesting a review.

Can you try again?

**Comment by readingdancer** [comment]
@leekelleher - Thanks for triggering that CoPilot review, based on it's feedback I've added some unit tests. I hope someone can take a look at this PR as I need the fix for a package I am building :) I have another similar PR in the pipeline for dropdowns and multi-select boxes, but it would be good to know if these will get approved / needs any changes before I submit the next one.

File: src/Umbraco.Web.UI.Client/src/packages/property-editors/checkbox-list/property-editor-ui-checkbox-list.element.ts [hunk 1]
```diff
@@ -89,6 +93,21 @@ export class UmbPropertyEditorUICheckboxListElement
...
+	 */
+	#updateCheckedState() {
+		if (this._list.length > 0) {
+			this._list = this._list.map(item => ({
+				...item,
+				checked: this.#selection.includes(item.value)
+			}));
+			// Trigger a re-render
+			this.requestUpdate();
+		}
```
[the remark this answers is not in the document]
Comment (readingdancer) [reply src/Umbraco.Web.UI.Client/src/packages/property-editors/checkbox-list/property-editor-ui-checkbox-list.element.ts:108]: I have submitted additional changes to fix these suggestions.

**Comment by readingdancer** [comment]
I am going to close this PR and open a new one for v16
